From a6113244523e53434fbfef1ddbbbf13020a8b811 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 15:50:22 -0700
Subject: [PATCH 36/96] HBASE-7560 TestCompactionState failures

Reason: Test failure
Author: Matteo Bertozzi
Ref: CDH-9858
(cherry picked from commit 80fb4a8d1a391676ea145ba0d5efe6a6e88e0a04)
---
 .../hbase/regionserver/TestCompactionState.java    |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionState.java b/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionState.java
index 79594d7..f1f32fd 100644
--- a/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionState.java
+++ b/src/test/java/org/apache/hadoop/hbase/regionserver/TestCompactionState.java
@@ -167,14 +167,11 @@ public class TestCompactionState {
           assertEquals(CompactionState.NONE, state);
         }
       } else {
-        curt = System.currentTimeMillis();
-        waitTime = 20000;
-        endt = curt + waitTime;
+        // Wait until the compaction is done
         state = admin.getCompactionState(table);
         while (state != CompactionState.NONE && curt < endt) {
           Thread.sleep(10);
           state = admin.getCompactionState(table);
-          curt = System.currentTimeMillis();
         }
         // Now, compaction should be done.
         assertEquals(CompactionState.NONE, state);
-- 
1.7.0.4

